invenio_documents/__init__.py

```diff
@@ -22,7 +22,122 @@
 # waive the privileges and immunities granted to it by virtue of its status
 # as an Intergovernmental Organization or submit itself to any jurisdiction.
 
-"""Invenio module for document management."""
+"""Modules that simplifies management of file URIs stored in metadata.
+
+Invenio-Documents is an integration layer between ``pyfilesystem``
+and ``Invenio-Records`` libraries. It handles URIs stored in JSON metadata
+specified using ``jsonpointer``.
+
+Documents proxy following operations:
+
+- `open`: for working with file handler.
+- `setcontents`: given a source file write its content.
+- `move`, `copy`, `remove`: simple proxies for standard file operations
+  while keeping the stored URI updated.
+
+Initialization
+--------------
+
+First create a Flask application (Flask-CLI is not needed for Flask
+version 1.0+):
+
+>>> from flask import Flask
+>>> from flask_cli import FlaskCLI
+>>> app = Flask('myapp')
+>>> app.config['SQLALCHEMY_DATABASE_URI'] = 'sqlite://'
+>>> ext_cli = FlaskCLI(app)
+
+You initialize Documents like a normal Flask extension, however Invenio-Documents
+depends on Invenio-Records which also depends on Invenio-DB so you need to
+initialize following extensions:
+
+>>> from invenio_db import InvenioDB
+>>> from invenio_records import InvenioRecords
+>>> from invenio_documents import InvenioDocuments
+>>> ext_db = InvenioDB(app)
+>>> ext_records = InvenioRecords(app)
+>>> ext_documents = InvenioDocuments(app)
+
+In order for the following examples to work, you need to work within an
+Flask application context so let's push one:
+
+>>> ctx = app.app_context()
+>>> ctx.push()
+
+Also, for the examples to work we need to create the database and tables (note,
+in this example we use an in-memory SQLite database):
+
+>>> from invenio_db import db
+>>> db.create_all()
+
+Document
+--------
+
+Document provides easy access to a file handler stored as URI inside a record
+metadata.
+
+The example will work with following metadata record:
+
+>>> from invenio_records.api import Record
+>>> record = Record.create(
+... {'title': 'Test', 'files': [{'uri': '/tmp/hello.txt'}]})
+
+
+Creating Documents
+~~~~~~~~~~~~~~~~~~
+
+First we need a pointer to URI stored in metadata.
+
+>>> uri_pointer = '/files/0/uri'
+
+With given ``uri_pointer`` we can instantiate ``Document`` class
```

Comment:
With given `the`  ``uri_pointer`` we can instantiate `a` ``Document`` class
and write something to it.